src/openforms/registrations/contrib/json_dump/tests/test_backend.py

```diff
@@ -0,0 +1,238 @@
+
+VCR_TEST_FILES = Path(__file__).parent / "files"
+
+
+class JSONDumpBackendTests(OFVCRMixin, TestCase):
+    VCR_TEST_FILES = VCR_TEST_FILES
+
+    def test_submission_happy_flow(self):
+        submission = SubmissionFactory.from_components(
+            [
+                {"key": "firstName", "type": "textfield"},
+                {"key": "lastName", "type": "textfield"},
+                {"key": "file", "type": "file"},
+            ],
+            completed=True,
+            submitted_data={
+                "firstName": "We Are",
+                "lastName": "Checking",
+                "file": [
+                    {
+                        "url": "some://url",
+                        "name": "my-foo.bin",
+                        "type": "application/foo",
+                        "originalName": "my-foo.bin",
+                    }
+                ],
+            },
+            bsn="123456789",
+        )
+
+        SubmissionFileAttachmentFactory.create(
+            form_key="file",
+            submission_step=submission.submissionstep_set.get(),
+            file_name="test_file.txt",
+            original_name="test_file.txt",
+            content_type="application/text",
+            content__data=b"This is example content.",
+            _component_configuration_path="components.2",
+            _component_data_path="file",
+        )
+
+        json_form_options = dict(
+            service=(ServiceFactory(api_root="http://localhost:80/")),
+            path="json_plugin",
+            variables=["firstName", "file", "auth_bsn"],
+        )
+        json_plugin = JSONDumpRegistration("json_registration_plugin")
+        set_submission_reference(submission)
+
+        expected_response = {
+            # Note that `lastName` is not included here as it wasn't specified in the variables
+            "data": {
+                "values": {
+                    "auth_bsn": "123456789",
+                    "file": {
+                        "file_name": "test_file.txt",
+                        "content": "VGhpcyBpcyBleGFtcGxlIGNvbnRlbnQu",
+                    },
+                    "firstName": "We Are",
+                },
+                "schema": {
+                    "$schema": "https://json-schema.org/draft/2020-12/schema",
+                    "type": "object",
+                    "properties": {
+                        "static_var_1": {"type": "string", "pattern": "^cool_pattern$"},
+                        "form_var_1": {"type": "string"},
+                        "form_var_2": {"type": "string"},
+                        "attachment": {"type": "string", "contentEncoding": "base64"},
+                    },
+                    "required": ["static_var_1", "form_var_1", "form_var_2"],
+                    "additionalProperties": False,
+                },
+            },
+            "message": "Data received",
+        }
+
+        res = json_plugin.register_submission(submission, json_form_options)
```

Comment:
```suggestion
        result = json_plugin.register_submission(submission, json_form_options)
```
let's use proper variable names